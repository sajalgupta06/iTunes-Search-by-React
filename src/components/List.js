// @flow
import React, { PropTypes } from 'react';
import Item from './Item';
import type { SearchResult } from '../type';
import '../style/List.css';

const List = ({
  results,
  resultCount
}: {
  results: Array<SearchResult>,
  resultCount: number
}) => (
  <div className="list-wrapper">
    {resultCount > 0 ? results.map((item, i) => <Item key={item.trackId || i} {...item} />) : null}
  </div>
);

export default List;
