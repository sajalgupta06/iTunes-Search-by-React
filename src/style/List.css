.list-wrapper {
  padding: 10px 45px;
  /* column width */
  column-width: 300px;
  /* space between columns */
  column-gap: 2px;
}
