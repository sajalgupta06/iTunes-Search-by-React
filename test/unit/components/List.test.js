import React from 'react';
import { shallow } from 'enzyme';
import List from '../../../src/components/List';

describe('<List />', () => {
  test('List component should renders correctly', () => {
    const props = {
      resultCount: 2,
      results: [
        {
          trackId: '1',
          trackPrice: 0.99,
          artworkUrl100: '',
          trackName: '',
          description: ''
        },
        {
          trackId: '2',
          trackPrice: 1.99,
          artworkUrl100: '',
          trackName: '',
          description: ''
        }
      ]
    };
    const component = shallow(<List {...props} />);
    expect(component).toMatchSnapshot();
  });

  test('no results', () => {
    const props = {
      resultCount: 0,
      results: []
    };
    const component = shallow(<List {...props} />);
    expect(component).toMatchSnapshot();
  });
});
