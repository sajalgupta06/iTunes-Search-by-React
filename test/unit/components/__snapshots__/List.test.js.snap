// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<List /> List component should renders correctly 1`] = `
<div
  className="list-wrapper"
>
  <Item
    artworkUrl100=""
    description=""
    key="1"
    trackId="1"
    trackName=""
    trackPrice={0.99}
  />
  <Item
    artworkUrl100=""
    description=""
    key="2"
    trackId="2"
    trackName=""
    trackPrice={1.99}
  />
</div>
`;

exports[`<List /> no results 1`] = `
<div
  className="list-wrapper"
/>
`;
